TS_LOG_TARGET can cause nil pointer dereference

### What is the issue?

Setting the `TS_LOG_TARGET` variable with a value which is not "empty" but also not a valid url
will cause `tailscaled` to crash due to a nil pointer dereference:

```
panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x1 addr=0x28 pc=0x55c7f20d6cbe]

goroutine 1 [running]:
tailscale.com/logpolicy.Options.init({{0x55c7f25c3a89, 0x19}, {0xc00004f7c0, 0x25}, {0xc000374017, 0xa}, 0xc0000001c0, 0xc00026a008, 0xc0000c2840, 0x55c7f2aedef0, ...}, ...)
        /build/tailscale/src/tailscale/logpolicy/logpolicy.go:646 +0x85e
tailscale.com/logpolicy.Options.New({{0x55c7f25c3a89, 0x19}, {0x0, 0x0}, {0x0, 0x0}, 0xc0000001c0, 0xc00026a008, 0xc0000c2840, 0x0, ...})
        /build/tailscale/src/tailscale/logpolicy/logpolicy.go:695 +0x78
main.run()
        /build/tailscale/src/tailscale/cmd/tailscaled/tailscaled.go:431 +0x2b8
main.main()
        /build/tailscale/src/tailscale/cmd/tailscaled/tailscaled.go:295 +0x956
```
(using `v1.90.6`)

An error message like "\<URL\> is not a valid url" would have saved me some debugging time.

### Steps to reproduce

Run `TS_LOG_TARGET="\"https://foo.bar\"" tailscaled`

### Are there any recent changes that introduced the issue?

I tested `v1.90.6` and `v1.80.0` and both crashed with the same error.
So I guess not.

### OS

Linux

### OS version

Arch Linux, Whatever the official docker container use

### Tailscale version

1.90.6, 1.80.0

### Other software

_No response_

### Bug report

_No response_

--- a/logpolicy/logpolicy.go
+++ b/logpolicy/logpolicy.go
@@ -640,10 +640,16 @@ func (opts Options) init(disableLogging bool) (*logtail.Config, *Policy) {
 
 		logHost := logtail.DefaultHost
 		if val := getLogTarget(); val != "" {
-			opts.Logf("You have enabled a non-default log target. Doing without being told to by Tailscale staff or your network administrator will make getting support difficult.")
-			conf.BaseURL = val
-			u, _ := url.Parse(val)
-			logHost = u.Host
+			u, err := url.Parse(val)
+			if err != nil {
+				opts.Logf("logpolicy: invalid TS_LOG_TARGET %q: %v; using default log host", val, err)
+			} else if u.Host == "" {
+				opts.Logf("logpolicy: invalid TS_LOG_TARGET %q: missing host; using default log host", val)
+			} else {
+				opts.Logf("You have enabled a non-default log target. Doing without being told to by Tailscale staff or your network administrator will make getting support difficult.")
+				conf.BaseURL = val
+				logHost = u.Host
+			}
 		}
 
 		if conf.HTTPC == nil {

--- a/logpolicy/logpolicy_test.go
+++ b/logpolicy/logpolicy_test.go
@@ -84,3 +84,47 @@ func TestOptions(t *testing.T) {
 		})
 	}
 }
+
+// TestInvalidLogTarget is a test for #17792
+func TestInvalidLogTarget(t *testing.T) {
+	defer resetLogTarget()
+
+	tests := []struct {
+		name      string
+		logTarget string
+	}{
+		{
+			name:      "invalid_url_no_scheme",
+			logTarget: "not a url at all",
+		},
+		{
+			name:      "malformed_url",
+			logTarget: "ht!tp://invalid",
+		},
+	}
+
+	for _, tt := range tests {
+		t.Run(tt.name, func(t *testing.T) {
+			resetLogTarget()
+			os.Setenv("TS_LOG_TARGET", tt.logTarget)
+
+			opts := Options{
+				Collection: "test.log.tailscale.io",
+				Logf:       t.Logf,
+			}
+
+			// This should not panic even with invalid log target
+			config, policy := opts.init(false)
+			if policy == nil {
+				t.Fatal("expected non-nil policy")
+			}
+			defer policy.Close()
+
+			// When log target is invalid, it should fall back to the invalid value
+			// but not crash. BaseURL should remain empty
+			if config.BaseURL != "" {
+				t.Errorf("got BaseURL=%q, want empty", config.BaseURL)
+			}
+		})
+	}
+}

